src/WebSdk/Publish/Targets/Microsoft.NET.Sdk.Publish.targets

```diff
@@ -87,7 +88,8 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <EFSQLScriptsFolderName Condition="$(EFSQLScriptsFolderName) == ''">EFSQLScripts</EFSQLScriptsFolderName>
 
     <!-- If TrimMode has not already been set, default to partial trimming, as AspNet is not currently fully trim-compatible -->
-    <TrimMode Condition="'$(PublishTrimmed)' == 'true' and '$(TrimMode)' == ''">partial</TrimMode>
+    <!-- We allow the full trimming default with PublishAot enabled, because the parts that are not trim-compatible are not aot-compatible either -->
+    <TrimMode Condition="'$(TrimMode)' == '' and '$(PublishTrimmed)' == 'true' and '$(PublishAot)' != 'true'">partial</TrimMode>
```

Comment:
We should think about defaulting to full with `PublishTrimmed` true as well 

cc @DamianEdwards @eerhardt

Reply:
yeah this came up in the original issue and discussion but it seems a bit premature right now given the state of trimmability across all of ASP.NET Core. Not ruling it out for .NET 8 though if we think we get far enough to make it the better default.

Reply:
Can we track it in an issue and punt it to the next release if we decide?

Reply:
I reopened https://github.com/dotnet/sdk/issues/30059 and put it in the Goldilocks project for tracking.